### PR TITLE
enhancement: Adding Github action to build and push custom catalog from feature branch

### PR DESCRIPTION
**Pull request by CFSNM**
## Description

This PR adds a new Github action to build and push catalog image to quay.io registry related to feature branches, in order to have the most updated catalog 

https://issues.redhat.com/browse/RHOAIENG-16343

## How Has This Been Tested?
This has been tested from local, and needs to be merged in order to test it using the pull_request_target kw

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/feature-branch-image.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    timeout-minutes: 60
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Login to quay.io
+        run: |
+          podman login -u ${{ secrets.QUAY_ID }} -p ${{ secrets.QUAY_TOKEN }} quay.io
+      - name: Build Feature Branch Catalog
+        run: |
+          make catalog-build -e CATALOG_IMG=${{ env.CATALOG_IMG }} -e BUNDLE_IMG=${{ env.BUNDLE_IMG }}
```
Comment (ykaliuta) [inline .github/workflows/feature-branch-image.yml:25]: Should you either use -e or CATALOG_IMG/BUNDLE_IMG from environment?

File: .github/workflows/feature-branch-image.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+name: Build and Push Feature Branch Custom Catalog
+env:
+  BUNDLE_IMG: quay.io/${{ secrets.QUAY_ORG }}/opendatahub-operator:${GITHUB_REF##*/}
+  CATALOG_IMG: quay.io/${{ secrets.QUAY_ORG }}/opendatahub-operator-catalog:${GITHUB_REF##*/}
+on:
+  push:
+    branches:
+      - "feature-**"
```
Comment (ykaliuta) [inline .github/workflows/feature-branch-image.yml:9]: Are we really targeting subdirectories here?

File: .github/workflows/feature-branch-image.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+name: Build and Push Feature Branch Custom Catalog
+env:
+  BUNDLE_IMG: quay.io/${{ secrets.QUAY_ORG }}/opendatahub-operator:${GITHUB_REF##*/}
```
Comment (zdtsw) [inline .github/workflows/feature-branch-image.yml:4]: isnt bundle in quay.io/${{ secrets.QUAY_ORG }}/opendatahub-operator-bundle:${GITHUB_REF##*/}

File: .github/workflows/feature-branch-image.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  push:
+    branches:
+      - "feature-**"
+permissions:
+  pull-requests: read
+  checks: read
+jobs:
+  build-and-push-feature-branch-custom-catalog:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
```
Comment (zdtsw) [inline .github/workflows/feature-branch-image.yml:16]: too much, 5mins enough

**Comment by CFSNM** [comment]
Closing this one, the change will be done in Openshift CI

**Pull request closed without merging**